spec/uploaders/alert_image_uploader_spec.rb

```diff
@@ -0,0 +1,44 @@
+require "rails_helper"
+require "carrierwave/test/matchers"
+
+RSpec.describe AlertImageUploader do
+  include CarrierWave::Test::Matchers
+
+  context "given a non-image or invalid base image" do
+    it "raises AlertImageGeneratorError" do
+      stolen_record = FactoryBot.create(:stolen_record, :with_bike_image)
+
+      image_assignment = -> do
+        image_path = stolen_record.bike_main_image.image.manipulate! { |img| img }
+        expect(image_path).to match(%r{tmp})
+      end
+
+      expect { image_assignment.call }.to raise_error(CarrierWave::ProcessingError)
+    end
+  end
+
+  context "given a valid bike image" do
+    let(:bike_with_valid_image) do
+      FactoryBot.create(:stolen_bike).tap do |bike|
```

Comment:
Good solution, but definitely took me a second to parse why it was useful.

Reply:
Which part? If you mean the file-in-let / close-in-after-each combination, I'm thinking of refactoring the other image-related tests toward something similar. Opening files in factories without closing them can lead to exceptions (so far they're not too common though.

```rb
# spec/uploaders/alert_image_uploader_spec.rb L27-29 

# define in a let in order to close file handler during teardown
let(:valid_image) { File.open(Rails.root.join("app/assets/images/pandabike.jpg"), "r") }
after(:each) { valid_image.close }
```

Reply:
The `tap`.

Reply:
Ah good nudge, thanks -- somewhat vestigial. Simplified in e06e4f6